### ISSUE TITLE
Make the parser/compiler usable under Node.js (in a nice way).

### ISSUE THREAD

**Issue by ChALkeR** [qmlweb/qmlweb#41]
This means:
- [x] Using proper `module.exports` instead of setting globals.
- [x] Making sure that we don't set globals in the parser.
- [x] Making an entrance point that allows users to do something as simple as:
 
 ```javascript
 const qmlweb = require('qmlweb');
 const data1 = qmlweb.parsePath(path_to_file);
 const data2 = qmlweb.parseQML(qml_file_content);
 const data3 = qmlweb.parseJS(js_file_content);
```

**Comment by labsin** [comment]
Should we add a UMD? Something like github.com/umdjs/umd/blob/master/templates/returnExports.js

**Comment by henrikrudstrom** [comment]
@ChALkeR Yeah, i was getting annoyed at all the globals floating around.
Think it would be good to make a separate module for the parser/loading stuff, 'qmlweb-parse' for example, then this module can be required by the gulp task or future server side needs.

@labsin UMD looks good. Are there any implications except the extra code to wrap around?

**Comment by ChALkeR** [comment]
@henrikrudstrom Actually, I think we can just use CJS and Browserify/Webpack. I will take a look at this soon.

**Comment by Plaristote** [comment]
Checkout the `browserified` branch to see a few tweaks I've done to make qmlweb usable directly from nodejs.
The encapsulate helpers are gone, there aren't any global variables left (apart from the AutoLoader, which might need to move somewhere else, since it's useless for the nodejs environment).

I've made some exports, but mainly I've cut the access from the external world to most of what was stored in global variables. We should discuss on what gets exported and what doesn't.

We might also consider removing most of the gulp concatenation in favor of uses of `require`. That would be a much harder piece of work though, and not being an expert in node's require function, I'm not completely sure this is the best approach.
Anyhow, it would definitely help in making qmlweb more modular.

I can make a PR if you figure what I did in `browserified` was the way to go.
I'm not sure about the last step. I had to generate two `lib/qt.js`: one for nodejs, and one that takes the nodejs version and uses browserify on it.
